frontend/src/components/SearchComponent.jsx

```diff
@@ -0,0 +1,31 @@
+import { Box, Button, Input } from "@mui/material";
+import SearchIcon from '@mui/icons-material/Search';
+
+const SearchComponent=()=>{
+  const handleSearch = async () => {
+    if (!map) return;
+    try {
+      const response = await fetch(`https://nominatim.openstreetmap.org/search?format=json&q=${encodeURIComponent(searchQuery)}`);
+      const data = await response.json();
+      if (data.length > 0) {
+        const { lat, lon } = data[0];
+        const center = fromLonLat([parseFloat(lon), parseFloat(lat)]);
+        setSelectedPlaceCoordinates(center);
+        map.getView().animate({ center, zoom: 10, duration: 1000 });
```

Comment:
As previously recommended in this code review, it's best to steer clear of literals, such as the numbers utilized in the _animate_ function. While it may not pose a significant issue in this instance, where it's the sole animation employed, it's important to consider future scenarios where multiple animations may be involved in a single application. In such cases, it's prudent to have the ability to control the maximum duration of all animations through a single configuration file, particularly if resource constraints are a concern.